packages/apps/reputation-oracle/server/src/modules/exchange/mexc-exchange.service.ts

```diff
@@ -0,0 +1,109 @@
+import { createHmac } from 'node:crypto';
+
+import { Inject, Injectable, forwardRef } from '@nestjs/common';
+
+import { StakingConfigService } from '@/config';
+import logger from '@/logger';
+import { ExchangeApiKeysService } from '@/modules/exchange-api-keys';
+
+import { ExchangeCredentials, ExchangeService } from './exchange.service';
+
+const MEXC_API_BASE_URL = 'https://api.mexc.com/api/v3';
+
+@Injectable()
+export class MexcExchangeService extends ExchangeService {
+  readonly id = 'mexc' as const;
+  readonly recvWindow = 5000;
+  private readonly logger = logger.child({ context: MexcExchangeService.name });
+
+  constructor(
+    @Inject(forwardRef(() => ExchangeApiKeysService))
+    private readonly exchangeApiKeysService: ExchangeApiKeysService,
+    private readonly stakingConfigService: StakingConfigService,
+  ) {
+    super();
+  }
+
+  checkRequiredCredentials(creds: ExchangeCredentials): boolean {
+    return Boolean(creds?.apiKey && creds?.secret);
+  }
+
+  async checkRequiredAccess(creds: ExchangeCredentials): Promise<boolean> {
+    const path = '/account';
+    const timestamp = Date.now();
+    const query = `timestamp=${timestamp}&recvWindow=${this.recvWindow}`;
+    const signature = createHmac('sha256', creds.secret)
+      .update(query)
+      .digest('hex');
+    const url = `${MEXC_API_BASE_URL}${path}?${query}&signature=${signature}`;
+
+    const controller = new AbortController();
+    const timeout = setTimeout(
+      () => controller.abort(),
+      this.stakingConfigService.timeoutMs,
+    );
+    try {
+      const res = await fetch(url, {
+        method: 'GET',
+        headers: { 'X-MEXC-APIKEY': creds.apiKey },
+        signal: controller.signal,
+      } as RequestInit);
+      return res.ok;
+    } catch (error) {
```

Comment:
Probably we should distinguish somehow if we got access error or some sort of network error, so we can distinguish between those and let the end use know